### ISSUE TITLE
Missing operator +() implementation after : String: compatibility with 64 bits scalars (#7863)

### ISSUE THREAD

**Issue by blue-wind-25** [esp8266/Arduino#7885]
### Basic Infos

- [X] This issue complies with the [issue POLICY doc](https://github.com/esp8266/Arduino/blob/master/POLICY.md).
- [X] I have read the documentation at [readthedocs](https://arduino-esp8266.readthedocs.io/en/latest) and the issue is not addressed there.
- [X] I have tested that the issue is present in current master branch (aka latest git).
- [X] I have searched the issue tracker for a similar issue.
- [ ] If there is a stack dump, I have decoded it.
- [X] I have filled out all fields below.

### Platform

- Hardware:					[All]
- Core Version:			[483519f852d37abbd4fc15063b60e9763ccc62c2]
- Development Env:		[All]
- Operating System:		[All]

### Settings in IDE

- Module:  			[All]
- Flash Mode:			[N/A]
- Flash Size: 		[N/A]
- lwip Variant:   [N/A]
- Reset Method:		[N/A]
- Flash Frequency:	[N/A]
- CPU Frequency:		[N/A]
- Upload Using:		[N/A]
- Upload Speed:   [N/A]

### Problem Description

In WString.h there are new friend functions:

       friend StringSumHelper &operator +(const StringSumHelper &lhs, long long num);
       friend StringSumHelper &operator +(const StringSumHelper &lhs, unsigned long long num);

The function implementations seem to be missing from WString.cpp

**Comment by earlephilhower** [comment]
@blue-wind-25 do you have a snippet that causes a compilation error?  Please post it so we have a good MCVE.

**Comment by blue-wind-25** [comment]
@earlephilhower no, because those functions are not actually used by any of the Arduino esp8266 core or library code.

I follow the repository because I use it as a refererence in my customized library that use different built system.

When I browsed the Arduino commit change, I found that those functions are declared, but seems to be not defined anywhere. So, it is not an actual bug report, but more like an information. I am sorry if this is the wrong place for this kind of information.
